pkg/ext/stores/tokens/tokens.go

```diff
@@ -1433,24 +1518,22 @@ func setExpired(token *ext.Token) error {
 	return nil
 }
 
-// ensureNameOrGenerateName ensures that the token has either a proper name, or
-// a generateName clause. Note, this function does __not generate__ the name if
-// the latter is present. That is delegated to the backend store, i.e. the
-// secrets holding tokens. See `secretFromToken` above.
-func ensureNameOrGenerateName(token *ext.Token) error {
-	// NOTE: When both name and generateName are set the generateName has precedence
-	if token.ObjectMeta.GenerateName != "" {
-		token.ObjectMeta.Name = ""
-		return nil
+// ensureNoNameAndGenerateName ensures that the token has the generateName
+// clause set, and no name specified. This enforces our decision (RFD Imperative
+// API Object Naming) to not allow users the choice of token names.
+func ensureNoNameAndGenerateName(token *ext.Token) error {
+	if token.ObjectMeta.Name != "" {
+		return apierrors.NewBadRequest(
+			"Token is invalid: metadata.name: Locked by system. Do not set.")
 	}
 
-	if token.ObjectMeta.Name != "" {
-		return nil
+	if token.ObjectMeta.GenerateName != "" && token.ObjectMeta.GenerateName != "token-" {
+		return apierrors.NewBadRequest(
+			"Token is invalid: metadata.generateName: Locked by system. Do not set.")
```

Comment:
Maybe we should just need to ignore these, as mentioned in https://github.com/rancher/rancher/pull/50207#discussion_r2084726547

Reply:
I am not sure if that would be good UX. User specifies things, is ignored, wonders what happened.

Reply:
made this so during the big save/restore rewrite.